torch/fx/experimental/symbolic_shapes.py

```diff
@@ -249,6 +249,16 @@ def _sympystr(self, printer):
 
         @classmethod
         def eval(cls, base, divisor):
+            # Python supports more types (like float), but we don't do type
+            # promotion here. And it's not clear whether we need to support more
+            # types based on existing models.
+            def check_supported_type(name, x):
+                if not isinstance(x, FloorDiv) and not x.is_integer:
+                    raise NotImplementedError(f"{name} must be an integer, but got: {type(x)}")
+
+            check_supported_type("base", base)
+            check_supported_type("divisor", divisor)
```

Comment:
perhaps not creating these strings is better and just throw a generic error msg?